src/main/java/echoserver/ClientConnection.java

```diff
@@ -4,34 +4,39 @@
 import java.net.Socket;
 
 public class ClientConnection {
-    public ClientConnection(){};
 
     public ClientConnection(Socket socket) {
         this.socket = socket;
     }
 
     public String readInput() throws IOException {
-        String connectionInput = convertInputStreamToString(socket.getInputStream());
-        return connectionInput;
+        return convertInputStreamToString(socket.getInputStream());
     }
 
     public void write(String message) throws IOException {
         message += "\n";
         socket.getOutputStream().write(message.getBytes());
     }
 
-    public void close() throws IOException {
-        this.socket.shutdownOutput();
-        this.socket.shutdownInput();
-        this.socket.close();
+    public void close() {
+        try {
+            EOFReached = true;
+            this.socket.shutdownOutput();
+            this.socket.shutdownInput();
+            this.socket.close();
+        }catch(IOException ignored) {}
     }
 
+    boolean EOFReached = false;
```

Comment:
I see that this is a public field, but other classes are using except for a test. Should this be public?